### DISCUSSION TITLE
Plans to support mapInPandas

### DISCUSSION THREAD

**Discussion by dbeavon** [dotnet/spark#915] [Q&A] [answered]
Are there plans to support this interface?
https://spark.apache.org/docs/3.0.0-preview/api/python/pyspark.sql.html#pyspark.sql.DataFrame.mapInPandas

Would that work similar to **GroupBy/Apply** (with RelationalGroupedDataset)?  Hopefully it would allow the remote UDF's to interact with the Apache Arrow recordbatch for inputs and outputs.

I can use **GroupBy/Apply** in the meantime.  But it seems like I'm always trying to find "artificial" ways of grouping the data just to be able to exchange recordbatch with the remote executors.    I believe the "artificial" groupings would become unnecessary if we could run this **mapInPandas**.

**Comment by imback82** [comment, marked as answer]
@dbeavon Could you create a feature request? I think we should be able to support this.

**Reply by dbeavon** [reply]
@imback82 Done.  I wasn't certain if this was already hidden somewhere in the existing code, or if it was a work in progress.

https://github.com/dotnet/spark/issues/925
